tests/test-team/template-mgmt-component-tests/template-mgmt-create-nhs-app-template-page.component.ts

```diff
@@ -43,4 +55,253 @@ test.describe('Create NHS App Template Page', () => {
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await createTemplatePage.clickContinueButton();
+    await expect(page.locator('.nhsuk-error-summary')).toBeVisible();
+    await expect(page.locator('.nhsuk-error-summary')).toHaveText([
+      'There is a problemEnter a template name',
+    ]);
+  });
+
+  test('404 Error returned when login link clicked', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+    await createTemplatePage.navigateToCreateNhsAppTemplatePage(
+      nhsAppNoTemplateSessionData.id
+    );
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/create-nhs-app-template/${nhsAppNoTemplateSessionData.id}`
+    );
+    await createTemplatePage.clickLoginLink();
+    await expect(page.locator('h1')).toHaveText('404');
+    await expect(page.getByText('This page could not be found')).toBeVisible();
+  });
+
+  test('5000 words Entered in Template body and word count correctly displayed', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.navigateToCreateNhsAppTemplatePage(
+      nhsAppNoTemplateSessionDataForInput.id
+    );
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/create-nhs-app-template/${nhsAppNoTemplateSessionDataForInput.id}`
+    );
+    const templateName = 'NHS Testing 123';
+    await page.locator('[id="nhsAppTemplateName"]').fill(templateName);
+    const templateMessage = 'T'.repeat(5000).trim();
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await expect(page.getByText('5000 of 5000 characters')).toBeVisible();
+  });
+
+  test('5001 words attempted to be entered in Template body and only 5000 allowed', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.navigateToCreateNhsAppTemplatePage(
+      nhsAppNoTemplateSessionDataForInput.id
+    );
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/create-nhs-app-template/${nhsAppNoTemplateSessionDataForInput.id}`
+    );
+    const templateName = 'NHS Testing 123';
+    await page.locator('[id="nhsAppTemplateName"]').fill(templateName);
+    const templateMessage = 'T'.repeat(5001).trim();
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await expect(page.getByText('5000 of 5000 characters')).toBeVisible();
+  });
+
+  test('Hyperlinks functionality', async ({ page, baseURL }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.navigateToCreateNhsAppTemplatePage(
+      nhsAppNoTemplateSessionData.id
+    );
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/create-nhs-app-template/${nhsAppNoTemplateSessionData.id}`
+    );
+    const footerLinks = [
+      {
+        name: 'Home',
```

Comment:
This name is wrong.  If we are going to check the links we should verify the href as well.  Also might be worth putting this data somewhere generic and reusable as they will be identical on every page.

Reply:
Ben implemented this so i'll update this test suite to use that.